internal/subscriber/subscriber_test.go

```diff
@@ -104,3 +108,94 @@ func TestSubscriber_MessageFiltering(t *testing.T) {
 	}
 
 }
+
+func TestSubscriber_Run(t *testing.T) {
+	ctx := context.Background()
+	_, cancelFunc := context.WithCancel(ctx)
+	defer cancelFunc()
+	ctrl := gomock.NewController(t)
+	requestChan := make(chan *PullRequest)
+	responseChan := make(chan *metrov1.PullResponse)
+	errChan := make(chan error, 1000)
+	closeChan := make(chan struct{})
+	cs := getMockConsumer(ctx, ctrl)
+	consumer := getMockConsumerManager(ctx, ctrl, cs)
+	type fields struct {
+		subscriber *Subscriber
+	}
+	subs := &Subscriber{
+		subscription:        getMockSubscription(),
+		topic:               topicName,
+		subscriberID:        subID,
+		requestChan:         requestChan,
+		responseChan:        responseChan,
+		ackChan:             make(chan *AckMessage, 10),
+		modAckChan:          make(chan *ModAckMessage, 10),
+		deadlineTicker:      time.NewTicker(1 * time.Second),
+		healthMonitorTicker: time.NewTicker(1 * time.Minute),
+		errChan:             errChan,
+		closeChan:           closeChan,
+		consumer:            consumer,
+		cancelFunc:          cancelFunc,
+		ctx:                 ctx,
+		retrier:             &retry.Retrier{},
+		subscriberImpl:      getMockBasicImplementation(ctx, consumer, ctrl),
+	}
+	type args struct {
+		ctx context.Context
+	}
+	tests := []struct {
+		name             string
+		fields           fields
+		args             args
+		maxNumOfMessages int32
+		expectedMsg      []string
+	}{
+		{
+			name: "Test Subscriber Run",
+			fields: fields{
+				subscriber: subs,
+			},
+			maxNumOfMessages: 1,
+			expectedMsg:      []string{"a", "b"},
+			args:             args{ctx: ctx},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cs.EXPECT().ReceiveMessages(gomock.Any(), messagebroker.GetMessagesFromTopicRequest{NumOfMessages: tt.maxNumOfMessages, TimeoutMs: 1000}).Return(
+				&messagebroker.GetMessagesFromTopicResponse{
+					Messages: getMockReceivedMessages(tt.expectedMsg),
+				}, nil)
+			sub := tt.fields.subscriber
+			go sub.Run(tt.args.ctx)
+			<-time.NewTicker(time.Second * 2).C
+			requestChan <- &PullRequest{
+				ctx:              ctx,
+				MaxNumOfMessages: 1,
+			}
+			select {
+			case res := <-responseChan:
+				assert.NotEmpty(t, res.ReceivedMessages)
+				assert.Equal(t, len(tt.expectedMsg), len(res.ReceivedMessages))
+				for _, msg := range res.ReceivedMessages {
+					ackMessage, _ := ParseAckID(msg.AckId)
+					ackMessage.ctx = ctx
+					sub.ackChan <- ackMessage
```

Comment:
why are we pushing to ack channel..? we are not consuming or doing anything on top of it..?

Reply:
Did not get you

Reply:
can you please explain?

Reply:
You are pushing a msg to ackChan, but you are not consuming that message from that channel or doing any assertions or anything..? Why exactly are we pushing a message here..?

Reply:
@sakshi-goyal-razorpay  as this will be duplicate test case, since in subImpl test case is already written for Acknowledge removing this